src/tbb/spin_mutex.cpp

```diff
@@ -39,6 +39,8 @@ void spin_mutex::scoped_lock::internal_release() {
 
 bool spin_mutex::scoped_lock::internal_try_acquire( spin_mutex& m ) {
     __TBB_ASSERT( !my_mutex, "already holding a lock on a spin_mutex" );
+    if( __TBB_IsLocked(m.flag) )
+        return false;
     bool result = bool( __TBB_TryLockByte(m.flag) );
```

Comment:
```suggestion
    bool result = !bool( __TBB_IsLocked(m.flag) ) && bool( __TBB_TryLockByte(m.flag) );
```

Reply:
While the suggested change appears technically accurate, can you elaborate if that change improve code-gen, performance, readability, or debuggability?  Or coding style standards related?

Reply:
It is more readable and makes the two changes symmetric:
### BEFORE
```
while( !__TBB_TryLockByte(flag) )
```
```
bool( __TBB_TryLockByte(m.flag) )
```
### AFTER
```
while (__TBB_IsLocked(flag) || !__TBB_TryLockByte(flag))
```
```
!bool( __TBB_IsLocked(m.flag) ) && bool( __TBB_TryLockByte(m.flag) )
```